### PR TITLE
Add second deep-dive XAI notebook

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
The second part of showing [Explainable AI toolkit](https://github.com/openvinotoolkit/openvino_xai) functionality.
First part: https://github.com/openvinotoolkit/openvino_notebooks/pull/2150

### CODE REVIEW

File: notebooks/explainable-ai-2-deep-dive/xai_classification.ipynb
Comment (negvet) [inline notebooks/explainable-ai-2-deep-dive/xai_classification.ipynb:1]: Align file name with the folder name?

**Review by negvet — CHANGES_REQUESTED** [review]
Please remove large cell outputs, it would be a bit easier to review.

I added targets property for explanation, please check it out (it basically returns `list(explanation.saliency_map.keys())`)

Do you mention somewhere that the model actually predicting `retriever_class_index = 206`?

At your basic usage, both preprocess_fn and postprocess_fn are optional, right?
If yes, this is not correct `Explainer will perform model inference but for that it requires preprocess_fn and postprocess_fn`

`targets=retriever_class_index` for single target

I propose to do overlay in the basic usage. And may be below show the case without overlay.

Let's skip info about the pre-postprocess functions. I mean what they might contain. This is a custom story and not related to the notebook. Just provide examples.

File: notebooks/explainable-ai-2-deep-dive/README.md
Comment (goodsong81) [inline notebooks/explainable-ai-2-deep-dive/README.md:1]: Can we add the short intro and links to the original algo references? (RISE, ReciproCAM, VITReciproCAM, ...) as we asked in the previous review?

**Comment by GalyaZalesskaya** [comment]
@negvet Some elaboration from my side:

`Do you mention somewhere that the model actually predicting retriever_class_index = 206?` - No, I ommited the inference call for simplicity, basically we can check saliency map for every class. I will mention that it's the predicted class, but won't add the code. Is it okay or it's better to be more transparent here?

`At your basic usage, both preprocess_fn and postprocess_fn are optional, right? If yes, this is not correct Explainer will perform model inference but for that it requires preprocess_fn and postprocess_fn` - actually, that's a question. If auto mode fails it runs black mode that requires postprocess mode. In basic mode, for simplicity we can not mention that and hope that white mode for chosen model will work. But I was afraid of issues that can appear with another model, so I mentioned that. I'll move this info from `Basic usage` to `pre-, postprocess_fn`, thank you.

`targets=retriever_class_index for single target` - I put it as a list to be able to update `targets` with more indexes easily without checking docs. I'll remove the list, but leave the comment, thank you.

**Comment by negvet** [comment]
> I will mention that it's the predicted class, but won't add the code. Is it okay or it's better to be more transparent here?

It is ok just to mention I think.

> But I was afraid of issues that can appear with another model, so I mentioned that.

I got it. It just was a bit confusing - functions are required buy actually not used in the example.
I agree, let's just move this elaboration down.

**Comment by GalyaZalesskaya** [comment]
<p>I wanted here just to show the command how to run explainer for all classes. Saliency maps are not visualized here for a sake of saving space. </p><p>Your comment is valid but it is better to show it in the 3rd interpretation notebook.</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2159/discussion/#comment-2209251546'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by GalyaZalesskaya** [comment]
<p>Great paragraph, thank you</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2159/discussion/#comment-2209251570'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by GalyaZalesskaya** [comment]
<p>The model in the notebook example doesn't have the named output so it fails then I access "logits". So I have to access logits as the first output of the model here</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2159/discussion/#comment-2209251585'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by goodsong81 — APPROVED** [review]

File: notebooks/explainable-ai-2-deep-dive/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+Using **OpenVINO XAI**, you can generate **saliency maps** that highlight regions of interest in input images from the model's perspective. This helps users understand why complex AI models produce specific responses.
+
+This notebook shows an example how to use OpenVINO XAI.
+
+It depicts a heatmap with areas of interest where neural network (classification or detection) focuses before making a decision.
+
+Example: Saliency map for `flat-coated retriever` class for MobileNetV3 classification model:
+
+![Saliency Map Example](./retriever-saliency-map.jpg)
```
Comment (eaidova) [inline notebooks/explainable-ai-2-deep-dive/README.md]: we prefer do not store images in repository (could you please remove?)
```suggestion
![Saliency Map Example](https://github.com/user-attachments/assets/5557d79d-2e9a-4784-aa17-fea2931a1435)
```

**Comment by GalyaZalesskaya** [comment]
@eaidova Thank you for your valuable feedback, made updates, please, take a look

**Review by goodsong81 — APPROVED** [review]

**Review by eaidova — APPROVED** [review]

**Comment by GalyaZalesskaya** [comment]
<p>Yes,the intention here is to see how these label names will look like: without the whole Imagenet name, that contains 3 options usually, with underscore. I think, it's useful, to know, which label name to expect</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2159/discussion/#comment-2227700940'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>
